File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_py/code_output/DataStatistics2.py
import math
from typing import List

class DataStatistics2:
    def __init__(self, data: List[int]) -> None:
        self.data = [float(x) for x in data]

    def getSum(self) -> float:
        return sum(self.data)

    def getMin(self) -> float:
        if not self.data:
            return float('nan')
        return min(self.data)

    def getMax(self) -> float:
        if not self.data:
            return float('nan')
        return max(self.data)

    def getVariance(self) -> float:
        if not self.data:
            return float('nan')
        mean = self.getSum() / len(self.data)
        return sum((x - mean) ** 2 for x in self.data) / len(self.data)

    def getStdDeviation(self) -> float:
        return math.sqrt(self.getVariance())

    def getCorrelation(self) -> float:
        return 1.0

File: data/derived_cot/rq2_eval/translation/qwen/_converted_code/java_to_cpp/DataStatistics2.cpp
#include <vector>
#include <limits>
#include <cmath>

class DataStatistics2 {
private:
    std::vector<double> data;

public:
    DataStatistics2(const std::vector<int>& data) {
        for (int val : data) {
            this->data.push_back(static_cast<double>(val));
        }
    }

    double get_sum() {
        if (data.empty()) {
            return std::numeric_limits<double>::quiet_NaN();
        }
        double sum = 0.0;
        for (double val : data) {
            sum += val;
        }
        return sum;
    }

    double get_min() {
        if (data.empty()) {
            return std::numeric_limits<double>::quiet_NaN();
        }
        double min_val = data[0];
        for (double val : data) {
            if (val < min_val) {
                min_val = val;
            }
        }
        return min_val;
    }

    double get_max() {
        if (data.empty()) {
            return std::numeric_limits<double>::quiet_NaN();
        }
        double max_val = data[0];
        for (double val : data) {
            if (val > max_val) {
                max_val = val;
            }
        }
        return max_val;
    }

    double get_variance() {
        if (data.empty()) {
            return std::numeric_limits<double>::quiet_NaN();
        }
        double mean = getSum() / data.size();
        double sqDiffSum = 0.0;
        for (double val : data) {
            double diff = val - mean;
            sqDiffSum += diff * diff;
        }
        return sqDiffSum / data.size();
    }

    double get_std_deviation() {
        if (data.empty()) {
            return std::numeric_limits<double>::quiet_NaN();
        }
        return std::sqrt(getVariance());
    }

    double get_correlation() {
        return 1.0;
    }
};